graphs/util/db.py

```diff
@@ -1608,6 +1608,298 @@ def find_all_graphs_containing_nodes(uid, search_type, search_word, view_type, d
 
 	return graph_dict.values()
 
+
+def uploadGPMLFile(username, graphJSON, title):
+	'''
+		Uploads GPML file to GraphSpace via /upload.
+		@param username: Owner of graph
+		@param graphJSON: JSON of graph
+		@param title: Title of graph
+	'''
+
+	try:
+		# Create JSON stucture for GraphSpace recognized JSON
+		parseJson = {"graph": {"edges": [], "nodes": []}, "metadata": {}}
+
+		# Use xml parser to parse the GPML file
+		import xml.etree.ElementTree as ET
+		tree = ET.fromstring(graphJSON)
+
+		# The current GPML format
+		GPML = '{http://pathvisio.org/GPML/2013a}'
+
+		# Keep the overall count of nodes and create unique ids for datanodes, shape, labels.
+		# IMPORTANT!!!
+		count = 0
+
+		# Parsing the GPML file for Shape elements
+		shapes = tree.findall(GPML+'Shape')
+		# Create a list of details of all the shape
+		shape_details = []
+		for shape in shapes:
+			tempNode = {}
+			tempNode.update(shape.attrib)
+			for more_details in shape:
+				tempNode.update(more_details.attrib)
+			shape_details.append(tempNode)
+
+		# Add shape elemets to the JSON structure parseJson
```

Comment:
You can add a method to add shape elements in the graphspace graph class. This method can be reused while doing the conversion from other formats as well.